contracts/PhasedEscrow.sol

```diff
@@ -0,0 +1,82 @@
+// ▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▌▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+//   ▓▓▓▓▓▓    ▓▓▓▓▓▓▓▀    ▐▓▓▓▓▓▓    ▐▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌   ▐▓▓▓▓▓▓
+//   ▓▓▓▓▓▓▄▄▓▓▓▓▓▓▓▀      ▐▓▓▓▓▓▓▄▄▄▄         ▓▓▓▓▓▓▄▄▄▄         ▐▓▓▓▓▓▌   ▐▓▓▓▓▓▓
+//   ▓▓▓▓▓▓▓▓▓▓▓▓▓▀        ▐▓▓▓▓▓▓▓▓▓▓         ▓▓▓▓▓▓▓▓▓▓         ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+//   ▓▓▓▓▓▓▀▀▓▓▓▓▓▓▄       ▐▓▓▓▓▓▓▀▀▀▀         ▓▓▓▓▓▓▀▀▀▀         ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▀
+//   ▓▓▓▓▓▓   ▀▓▓▓▓▓▓▄     ▐▓▓▓▓▓▓     ▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌
+// ▓▓▓▓▓▓▓▓▓▓ █▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+// ▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+//
+//                           Trust math, not hardware.
+
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+interface IRewardsPoolContract {
+    function topUpReward(uint256 amount) external;
+}
+
+// @title PhasedEscrow
+/// @notice A token holder contract allowing contract owner to set rewards pool
+///         of tokens held by the contract and allowing the owner to withdraw the
+///         tokens to that rewards pool in phases.
+contract PhasedEscrow is Ownable {
+    using SafeERC20 for IERC20;
+
+    event RewardsPoolUpdated(address rewardsPool);
+    event TokensWithdrawn(address rewardsPool, uint256 amount);
+
+    IERC20 public immutable token;
+    IRewardsPoolContract public rewardsPool;
+
+    constructor(IERC20 _token) {
+        token = _token;
+    }
+
+    /// @notice Sets the provided address as a rewards pool allowing it to
+    ///         withdraw all tokens from escrow. This function can be called only
+    ///         by escrow owner.
+    function setRewardsPool(IRewardsPoolContract _rewardsPool)
+        external
+        onlyOwner
+    {
+        rewardsPool = _rewardsPool;
+        emit RewardsPoolUpdated(address(rewardsPool));
+    }
+
+    /// @notice Withdraws the specified number of tokens from escrow to the
+    ///         rewards pool. If the rewards pool is not set, or there are
+    ///         insufficient tokens in escrow, the function fails.
+    function withdraw(uint256 amount) external onlyOwner {
+        require(
+            address(rewardsPool) != address(0),
+            "Rewards pool not assigned"
+        );
+
+        uint256 balance = token.balanceOf(address(this));
+        require(amount <= balance, "Not enough tokens for withdrawal");
+
+        emit TokensWithdrawn(address(rewardsPool), amount);
+
+        token.safeApprove(address(rewardsPool), amount);
+        rewardsPool.topUpReward(amount);
+    }
+
+    /// @notice Funds the escrow by transferring all of the approved tokens
+    ///         to the escrow.
+    function receiveApproval(
+        address _from,
+        uint256 _value,
+        address _token,
+        bytes memory
```

Comment:
`bytes` is unused here. Is this a standard to include it in `reveiveApproval` functions or can we remove it?

Reply:
Yes, this interface is an industry standard though I think it was never included in an EIP that got finalized.